libs/gl-testing/gltesting/fixtures.py

```diff
@@ -120,6 +121,15 @@ def mock_estimatesmartfee(r):
     del os.environ["GL_SCHEDULER_GRPC_URI"]
     s.stop()
 
+    # Check the reporting server to see if there were any failed
+    # request resolutions or policies. When testing we bail the test
+    # here.
+
+    if s.debugger.reports != []:
```

Comment:
Calling a ValueError on teardown if something is unexpected is pure gold! Learned something new today!

Reply:
Yeah, it's quite nice for invariant checking, and we do it a log in CLN itself. There is one caveat though: things like `flaky` will not rerun a test if it's a fixture that causes the error, rather than the test itself. So rerunning failures doesn't work for these post-mortem checks, and so we try to keep them to a minimum to avoid too much CI breakage.